src/provider_hydrophone_node.cc

```diff
@@ -0,0 +1,261 @@
+/**
+ * \file	provider_hydrophone_node.cc
+ * \author	Francis Alonzo
+ * \date	2021/10/20
+ *
+ * \copyright Copyright (c) 2021 S.O.N.I.A. All rights reserved.
+ *
+ * \section LICENSE
+ *
+ * This file is part of S.O.N.I.A. software.
+ *
+ * S.O.N.I.A. software is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * S.O.N.I.A. software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with S.O.N.I.A. software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "provider_hydrophone_node.h"
+
+#define MAX_BUFFER_SIZE 4096
+
+namespace provider_hydrophone {
+
+//==============================================================================
+// C / D T O R S   S E C T I O N
+
+//------------------------------------------------------------------------------
+//
+  ProviderHydrophoneNode::ProviderHydrophoneNode(const ros::NodeHandlePtr &nh)
+    : nh_(nh),
+      configuration_(nh),
+      serialConnection_(configuration_.getTtyPort())
+  {
+    //server.setCallback(boost::bind(&ProviderHydrophoneNode::CallBackDynamicReconfigure, this, _1, _2));
+
+    //HydroConfig config;
+    //CallBackDynamicReconfigure(config, 0);
+
+    pingPublisher_ = nh_->advertise<sonia_common::PingMsg>("/provider_hydrophone/ping", 100);
+
+    readerThread = std::thread(std::bind(&ProviderHydrophoneNode::readThread, this));
+    h1ParseThread = std::thread(std::bind(&ProviderHydrophoneNode::h1RegisterThread, this));
+
+    //setGain(current_gain_);
+  }
+
+  //------------------------------------------------------------------------------
+  //
+  ProviderHydrophoneNode::~ProviderHydrophoneNode() {
+      serialConnection_.~Serial();
+      readerThread.~thread();
+      h1ParseThread.~thread();
+  }
+
+  //==============================================================================
+  // M E T H O D   S E C T I O N
+  //------------------------------------------------------------------------------
+  //
+  void ProviderHydrophoneNode::Spin() 
+  {
+    ros::Rate r(10);  // 10 hz
+
+    startAcquireNormalMode();
+
+    while (ros::ok()) 
+    {
+      ros::spinOnce();
+      r.sleep();
+    }
+  }
+
+  /*void ProviderHydrophoneNode::CallBackDynamicReconfigure(provider_hydrophone::HydroConfig &config, uint32_t level)
+  {
+    ROS_INFO_STREAM("DynamicReconfigure callback. Old gain : " << gain_ << " new gain : " << config.Gain);
+    gain_ = config.Gain;
+  }*/
```

Comment:
I know we talked about it, but Dynamic Reconfigure can still be used with an old telemetry module.